### PR TITLE
[8.0][ADD] New module: product_supplierinfo_customer_code

### PR DESCRIPTION
**Pull request by zamberjo**
Inherits ``product_supplierinfo_for_customer`` and replicates sale line's name
field behaviour in ``product.supplierinfo``. Final name will be
``customer product code + name`` instead of ``product code + name``.

See PR #267

### CODE REVIEW

**Comment by eLBati** [comment]
@zamberjo thanks.
I would rename it to ```sale_order_customer_code``` and move to https://github.com/OCA/sale-workflow
cc @lasley @pedrobaeza 

@zamberjo could you also add tests?

File: product_supplierinfo_customer_code/README.rst [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+==============================
+Product customer info on sales
+==============================
+
+Inherits ``product_supplierinfo_for_customer`` and replicates sale line's name
+field behaviour in ``product.supplierinfo``. Final name will be
+``customer product code + name`` instead of ``product code + name``.
+
+Installation
+============
```
Comment (aheficent) [inline product_supplierinfo_customer_code/README.rst]: This is not needed.

File: product_supplierinfo_customer_code/README.rst [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+Known issues / Roadmap
+======================
+
+* ...
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues
+<https://github.com/OCA/{project_repo}/issues>`_. In case of trouble, please
```
Comment (aheficent) [inline product_supplierinfo_customer_code/README.rst]: Please adapt to the current repo

File: product_supplierinfo_customer_code/README.rst (hunk 1 above)
Comment (AaronHForgeFlow) [inline product_supplierinfo_customer_code/README.rst:14]: Please remove empty sections

File: product_supplierinfo_customer_code/README.rst (hunk 2 above)
Comment (AaronHForgeFlow) [inline product_supplierinfo_customer_code/README.rst:55]: Please adapt to the current repo

**Review by AaronHForgeFlow — COMMENTED** [review]
I agree with @eLBati  to rename this to sale_order_customer_code and move to https://github.com/OCA/sale-workflow

**Comment by AaronHForgeFlow** [comment]
@zamberjo  Sorry for the duplicated comments. Can anybody delete those?

**Comment by elicoidal** [comment]
@aheficent done

**Comment by zamberjo** [comment]
There exists a similar module that has been submitted recently (https://github.com/OCA/product-attribute/tree/8.0/product_supplierinfo_for_customer_sale), is this PR still necessary?

**Comment by AaronHForgeFlow** [comment]
@zamberjo I think that this module overlaps product_supplierinfo_for_customer_sale so I'd close this one.

**Pull request closed without merging**